packages/eas-cli/src/build/ios/version.ts

```diff
@@ -0,0 +1,128 @@
+import { ExpoConfig } from '@expo/config';
+import { IOSConfig } from '@expo/config-plugins';
+import chalk from 'chalk';
+import nullthrows from 'nullthrows';
+import semver from 'semver';
+
+import Log from '../../log';
+import { promptAsync } from '../../prompts';
+import { updateAppJsonConfigAsync } from '../utils/appJson';
+import { readPlistAsync, writePlistAsync } from './plist';
+
+export enum BumpStrategy {
+  SHORT_VERSION,
+  BUILD_NUMBER,
+  NOOP,
+}
+
+export async function bumpVersionAsync({
+  bumpStrategy,
+  projectDir,
+  exp,
+}: {
+  projectDir: string;
+  exp: ExpoConfig;
+  bumpStrategy: BumpStrategy;
+}): Promise<void> {
+  if (bumpStrategy === BumpStrategy.NOOP) {
+    return;
+  }
+  const infoPlist = await readInfoPlistAsync(projectDir);
+  await bumpVersionInAppJsonAsync({ bumpStrategy, projectDir, exp });
+  Log.log('Updated versions in app.json');
+  await writeVersionsToInfoPlistAsync({ projectDir, exp, infoPlist });
+  Log.log('Synchronized versions with Info.plist');
+}
+
+export async function bumpVersionInAppJsonAsync({
+  bumpStrategy,
+  projectDir,
+  exp,
+}: {
+  bumpStrategy: BumpStrategy;
+  projectDir: string;
+  exp: ExpoConfig;
+}): Promise<void> {
+  if (bumpStrategy === BumpStrategy.NOOP) {
+    return;
+  }
+  Log.addNewLineIfNone();
+  if (bumpStrategy === BumpStrategy.SHORT_VERSION) {
+    const shortVersion = IOSConfig.Version.getVersion(exp);
+    if (semver.valid(shortVersion)) {
+      const bumpedShortVersion = nullthrows(semver.inc(shortVersion, 'patch'));
+      Log.log(
+        `Bumping ${chalk.bold('expo.version')} from ${chalk.bold(shortVersion)} to ${chalk.bold(
+          bumpedShortVersion
+        )}`
+      );
+      await updateAppJsonConfigAsync({ projectDir, exp }, config => {
+        config.version = bumpedShortVersion;
+      });
+    } else {
+      Log.log(`${chalk.bold('expo.version')} = ${chalk.bold(shortVersion)} is not a valid semver`);
+      const { bumpedShortVersion } = await promptAsync({
+        type: 'text',
+        name: 'bumpedShortVersion',
+        message: 'What is the next version?',
+      });
+      await updateAppJsonConfigAsync({ projectDir, exp }, config => {
+        config.version = bumpedShortVersion;
+      });
+    }
+  } else {
+    const buildNumber = IOSConfig.Version.getBuildNumber(exp);
+    if (buildNumber.match(/^\d+$/)) {
```

Comment:
If I remember correctly we discussed that in the case of build number we require the last segment to be a number but except that we should support all formats allowed for this value.